tests/rustdoc-ui/2024-doctests-checks.rs

```diff
@@ -0,0 +1,41 @@
+//@ check-pass
+//@ compile-flags: --test --test-args=--test-threads=1 -Zunstable-options --edition 2024
+//@ normalize-stdout-test: "tests/rustdoc-ui" -> "$$DIR"
+//@ normalize-stdout-test "finished in \d+\.\d+s" -> "finished in $$TIME"
+//@ normalize-stdout-test ".rs:\d+:\d+" -> ".rs:$$LINE:$$COL"
+
+/// This one should fail: crate attributes should remain crate attributes
+/// in standalone doctests.
+///
+/// ```compile_fail
+/// #![deny(missing_docs)]
```

Comment:
The problem with this test is it would also (I believe) fail if it weren't a crate-level attribute. Maybe we should use `#![feature(..)]` or something like that instead?

Reply:
The problem with features is that they eventually get stabilized. ^^'

Reply:
Also: doctests with `features` are automatically excluded from merged doctests.

Reply:
Hmm, that's true. Aren't there some rustdoc attributes that can only be applied at the crate root though? Like playground_url or something.

Reply:
If there are crate attributes, it's excluded from merged doctests, so again, all good. :D

Reply:
I think we're miscommunicating :)

I know that they're excluded from merged doctests --  that's my point. I want this test to use an attribute that would error if it were not at the crate root, that way we can test that the merged doctests exclusion is working properly. `#![deny(missing_docs)]` will cause the test to fail compilation whether we merged it or not (so it's impossible to tell if it's working correctly). Meanwhile `html_playground_url` will pass compilation at the crate-level but cause an error at the module-level:

```
error: this attribute can only be applied at the crate level
 --> src/lib.rs:2:12
  |
2 |     #![doc(html_playground_url = "foo")]
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: read <https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#at-the-crate-level> for more information
  = note: `#[deny(invalid_doc_attributes)]` on by default

error: could not compile `playground` (lib) due to 1 previous error
```

Reply:
You'd of course also have to remove the `compile_fail` from the doctest.

Reply:
Thanks for the clarification! Added the test. :)